modin/utils.py

```diff
@@ -342,21 +389,21 @@ def _inherit_docstrings(parent, excluded=[], overwrite_existing=False, apilink=N
     which means that ancestor class attribute docstrings could also change.
     """
 
-    def _documentable_obj(obj):
+    def _documentable_obj(obj: object) -> bool:
         """Check if `obj` docstring could be patched."""
-        return (
+        return bool(
             callable(obj)
             or (isinstance(obj, property) and obj.fget)
             or (isinstance(obj, (staticmethod, classmethod)) and obj.__func__)
         )
 
-    def decorator(cls_or_func):
+    def decorator(cls_or_func: Fn) -> Fn:
         if parent not in excluded:
             _replace_doc(parent, cls_or_func, overwrite_existing, apilink)
 
         if not isinstance(cls_or_func, types.FunctionType):
             seen = set()
-            for base in cls_or_func.__mro__:
+            for base in cls_or_func.__mro__:  # type: ignore
```

Comment:
Why ignore?

Reply:
`__mro__` is only defined for classes, not functions, and is gated here by the `isinstance(cls_or_func, types.FunctionType)` check. The type annotation isn't strict enough for mypy to recognize that the variable must be a class in this branch, and I'm not sure what the correct annotation would be to get `__mro__` as a property.